hyperspy/_signals/hologram_image.py

```diff
@@ -0,0 +1,303 @@
+
+        w_obj = self._reconstruct(holo_data, sb_size, sb_pos, fresnel_width=fresnel_width)  # object wave
+
+        wave = w_obj / w_ref
+        wave_image = self._deepcopy_with_new_data(wave)
+        wave_image.set_signal_type('wave')  # New signal is normal image without special signal type!
+        rec_param_dict = {'sb_pos_x0': rec_param[0], 'sb_pos_x1': rec_param[1],
+                          'sb_pos_y0': rec_param[2], 'sb_pos_y1': rec_param[3],
+                          'sb_size': rec_param[4]}
+        wave_image.metadata.Signal.add_node('holo_rec_param')
+        wave_image.metadata.Signal.holo_rec_param.add_dictionary(rec_param_dict)
+        wave_image.axes_manager[0].scale = wave_image.axes_manager[0].scale*sx/wave.shape[0]
+        wave_image.axes_manager[1].scale = wave_image.axes_manager[1].scale*sy/wave.shape[1]
+        if show_phase:
+            phase = np.angle(wave)
+            f, ax = plt.subplots(1, 1)
+            ax.imshow(phase, cmap=cm.binary_r)
+            # f.canvas.manager.window.raise_()
+        if return_param:
+            return wave_image, rec_param
+        else:
+            return wave_image
+
+
+    @staticmethod
+    def _reconstruct(holo_data, sb_size, sb_pos, fresnel_ratio=0.3, fresnel_width=6):
+        """Core function for holographic reconstruction performing following steps:
+
+        * 2D FFT without apodisation;
+
+        * Cutting out sideband;
+
+        * Centering sideband;
+
+        * Applying round window;
+
+        * Applying sinc filter;
+
+        * Applying automatic filtering of Fresnel fringe (Fresnel filtering);
+
+        * Inverse FFT.
+
+        Parameters
+        ----------
+        holo_data : array_like
+            Holographic data array
+        sb_size : int
+            Size of the sideband filter in px
+        sb_pos : nparray (N=1)
+            Vector of two elements with sideband coordinates [y,x]
+        fresnel_ratio : float
+            The ratio of Fresnel filter with respect to the sideband size
+        fresnel_width : int
+            Width of fresnel fringe filter in px, set to 0 to disable filtering
+
+        Returns
+        -------
+            wav : nparray
+                Reconstructed electron wave
+
+        Notes
+        -----
+
+        Disabeling of Fresnel filter is not implemented
+
+        See Also
+        --------
+
+        reconstruct
+
+        """
+        # TODO: Parsing of the input has to be redone
+        # TODO: Add smoothing of Fresnel filter
+        # TODO: Add other smoothing options?
+
+        image_size = sb_size*2
+        (sx, sy) = holo_data.shape
+        holo_data = np.float64(holo_data)
+
+        h_hw_fft = fftshift(fft2(holo_data))  # <---- NO Hanning filtering
+
+        sb_roi = h_hw_fft[(sb_pos[0]-image_size//2):(sb_pos[0]+image_size//2),
+                 (sb_pos[1]-image_size//2):(sb_pos[1]+image_size//2)]
+
```

Comment:
> Apparently the test is still failing here, because the code tries to slice with floats. Can you debug this, to see where the float occurs? If I understood you correctly, on your system, the test does not fail, correct?
I'll do debugging. It should be easy to fix, but I would expect the error arising again in other places...
On my system I never got any problem with it - the tests are running. I'm puzzled here.